gtda/curves/_functions.py

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+from joblib import Parallel, delayed, effective_n_jobs
+from sklearn.utils import gen_even_slices
+
+
+_AVAILABLE_FUNCTIONS = {
+    'flatten': {},
+    'argmax': {},
+    'argmin': {},
+    'min': {},
+    'max': {},
+    'mean': {},
+    'std': {},
+    'median': {},
+    'average': {'weights': {'type': np.ndarray}},
+}
+
+_implemented_function_recipes = {
+    'flatten': lambda X, axis: np.reshape(X, (X.shape[0], -1)),
```

Comment:
The `axis` argument seems to be unused.

Reply:
On purpose. All the other takes one, so instead of redefining all other function with axis=-1, I hardcoded it in this lambda.

Reply:
Thanks for the explanation!